psycopg/psycopg/rows.py

```diff
@@ -207,6 +207,28 @@ def kwargs_row__(values: Sequence[Any]) -> T:
     return kwargs_row_
 
 
+def scalar_row(cursor: "BaseCursor[Any, Any]") -> "RowMaker[Any]":
+    """
+    Generate a row factory returning the scalar value
+    taken from the first and only column.
+    """
+    res = cursor.pgresult
+    if not res:
+        return no_result
+
+    nfields = _get_nfields(res)
+    if nfields is None:
+        return no_result
+
+    if nfields != 1:
+        raise e.ProgrammingError(f"Expected exactly one column, found {nfields}")
```

Comment:
I wonder if we shouldn't accept queries that return more than one column, and just discard the one after the first.

```suggestion

    if nfields < 1:
        raise e.ProgrammingError(f"at least one column expected")
```

If you are wondering if it's possible to write a query that returns zero columns, it is: just `SELECT`.

What do you think?

Regardless of this choice, the error message should start with lowercase (if it start with uppercase, it is a complete sentence, and should end with a full-stop. However it's easier to compose a message if it's lowercase and has no final full-stop).

Reply:
I don't have a strict opinion on this matter. I am OK with your suggestion.